core/src/main/java/tachyon/worker/BlockHandler.java

```diff
@@ -18,86 +18,82 @@
+
+import io.netty.channel.FileRegion;
 
 import tachyon.TachyonURI;
 
 /**
- * Base class for handling block I/O. Block handlers for different under file systems can be
- * implemented by extending this class. It is not thread safe, the caller must guarantee thread
- * safe. This class is internal and subject to changes.
+ * General interface for handling block I/O. Block handlers for different under file systems
+ * implement this interface. It is not thread safe, the caller must guarantee thread safe.
+ * This interface is internal and subject to changes.
  */
-public abstract class BlockHandler implements Closeable {
+public interface BlockHandler extends Closeable {
 
-  /**
-   * Create a block handler according to path scheme
-   * 
-   * @param path the block path
-   * @return the handler of the block
-   * @throws IOException
-   * @throws IllegalArgumentException
-   */
-  public static BlockHandler get(String path) throws IOException, IllegalArgumentException {
-    if (path.startsWith(TachyonURI.SEPARATOR) || path.startsWith("file://")) {
-      return new BlockHandlerLocal(path);
+  class Factory {
+
+    /**
+     * Create a block handler according to path scheme
+     * 
+     * @param path the path of the block
+     * @return the handler for the block
+     * @throws IOException
+     * @throws IllegalArgumentException
+     */
+    public static BlockHandler get(String path) throws IOException, IllegalArgumentException {
+      if (path.startsWith(TachyonURI.SEPARATOR) || path.startsWith("file://")) {
+        return new BlockHandlerLocal(path);
+      }
+      throw new IllegalArgumentException("Unsupported block file path: " + path);
     }
-    throw new IllegalArgumentException("Unsupported block file path: " + path);
   }
 
   /**
-   * Append data to the block from a byte array
+   * Delete the block
    * 
-   * @param blockOffset starting position of the block file
-   * @param buf the data buffer
-   * @param offset the offset of the buffer
-   * @param length the length of the data
-   * @return the size of data that was written
+   * @return true if success, otherwise false
    * @throws IOException
    */
-  public int append(long blockOffset, byte[] buf, int offset, int length) throws IOException {
-    return append(blockOffset, ByteBuffer.wrap(buf, offset, length));
-  }
+  boolean delete() throws IOException;
 
   /**
-   * Appends data to the block from a ByteBuffer
+   * Get file region for some part of block
    * 
-   * @param blockOffset starting position of the block file
-   * @param srcBuf ByteBuffer that data is stored in
-   * @return the size of data that was written
-   * @throws IOException
+   * @param position the starting position
+   * @param length the length of the region
+   * @return file region for the specific part of block
    */
-  public abstract int append(long blockOffset, ByteBuffer srcBuf) throws IOException;
+  FileRegion getFileRegion(long position, long length);
```

Comment:
Why do we include a netty specific method in the general BlockHandler interface?